### PR TITLE
Add AWS role field to OtpServer

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Re: ibi-group/datatools-server#272. This simply adds a field in ServerSettings to account for the backend changes. This depends on ibi-group/datatools-server#273

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
The updated documentation reads fine. Glad you included direct links to the AWS roles page.

File: docs/user/setting-up-aws-servers.md [hunk 1]
```diff
@@ -75,3 +76,70 @@ The load balancer also allows instantiating multiple OTP servers on large deploy
 2. Select `Create Record Set`. Fill in the subdmain (e.g. `otp-mod-dev.ibi-transit.com`). Set the `Record Type` to `CNAME`.
 3. In the value field, paste the `DNS Name` of the load balancer instance above.
 4. Click `Create`.
+
+## Delegate Third Party Account Access
+For Data Tools to access AWS resources (e.g., S3 and EC2) in third party AWS accounts, additional setup is required. The steps provided in [this AWS tutorial](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html) detail the process to delegate access to across AWS accounts. A more tailored, shorthand version of this process is provided below, but if more background information is needed about, please follow the tutorial link above.
```
Comment (evansiroky) [inline docs/user/setting-up-aws-servers.md]: Can get rid of "to" here: `access to across`

File: docs/user/add-deployment-server.md [hunk 2]
```diff
@@ -13,11 +13,12 @@ From `Administration > Deployment servers`, click on `+ Add Server`.
 | Property | Description |
 |----------|-------------|
 | Name     | A descriptive display name for the server.        |
-| Public URL | The URL where the public can access the Data Tools UI, e.g. `https://otp-mod-ui.ibi-transit.com`. It is typcially the  CNAME of the CloudFront mirror of the AWS S3 bucket you created or picked for this deployment.         |
+| Public URL | The URL where the public can access the Data Tools UI, e.g. `https://otp-mod-ui.ibi-transit.com`. It is typically the  CNAME of the CloudFront mirror of the AWS S3 bucket you created or picked for this deployment.         |
```
Comment (evansiroky) [inline docs/user/add-deployment-server.md]: Extra space here: `the  CNAME`

**Pull request merged**